src/pybind/mgr/devicehealth/module.py

```diff
@@ -220,6 +221,37 @@ def config_notify(self):
                     self.get_module_option(opt['name']))
             self.log.debug(' %s = %s', opt['name'], getattr(self, opt['name']))
 
+    def notify(self, notify_type, notify_id):
+       # create device_health_metrics pool if it doesn't exist
+       if notify_type == "osd_map":
+            if not self.has_device_pool:
+                self.create_device_pool()
```

Comment:
i think this change breaks `qa/workunits/mon/pg_autoscaler.sh`.

```
2020-09-16T17:41:54.662 INFO:tasks.workunit:Running workunit mon/pg_autoscaler.sh...
2020-09-16T17:41:54.662 INFO:teuthology.orchestra.run.smithi169:workunit test mon/pg_autoscaler.sh> mkdir -p -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && cd -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && CEPH_CLI_TEST_DUP_COMMAND=1 CEPH_REF=8f1d1e8dc6e05bdb60f8d0a748527c08b7173201 TESTDIR="/home/ubuntu/cephtest" CEPH_ARGS="--cluster ceph" CEPH_ID="0" PATH=$PATH:/usr/sbin CEPH_BASE=/home/ubuntu/cephtest/clone.client.0 CEPH_ROOT=/home/ubuntu/cephtest/clone.client.0 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 3h /home/ubuntu/cephtest/clone.client.0/qa/workunits/mon/pg_autoscaler.sh
2020-09-16T17:41:54.719 INFO:tasks.workunit.client.0.smithi169.stderr:++ ceph osd ls
2020-09-16T17:41:54.719 INFO:tasks.workunit.client.0.smithi169.stderr:++ wc -l
2020-09-16T17:41:55.099 INFO:tasks.workunit.client.0.smithi169.stderr:+ NUM_OSDS=8
2020-09-16T17:41:55.100 INFO:tasks.workunit.client.0.smithi169.stderr:+ '[' 8 -lt 6 ']'
2020-09-16T17:41:55.100 INFO:tasks.workunit.client.0.smithi169.stderr:++ ceph osd pool ls
2020-09-16T17:41:55.100 INFO:tasks.workunit.client.0.smithi169.stderr:++ wc -l
2020-09-16T17:41:55.472 INFO:tasks.workunit.client.0.smithi169.stderr:+ NUM_POOLS=1
2020-09-16T17:41:55.472 INFO:tasks.workunit.client.0.smithi169.stderr:+ '[' 1 -gt 0 ']'
2020-09-16T17:41:55.472 INFO:tasks.workunit.client.0.smithi169.stderr:+ echo 'test requires no preexisting pools'
2020-09-16T17:41:55.472 INFO:tasks.workunit.client.0.smithi169.stderr:+ exit 1
2020-09-16T17:41:55.472 INFO:tasks.workunit.client.0.smithi169.stdout:test requires no preexisting pools
2020-09-16T17:41:55.473 DEBUG:teuthology.orchestra.run:got remote process result: 1
```

see https://pulpito.ceph.com/kchai-2020-09-16_14:28:48-rados-wip-kefu-testing-2020-09-16-1741-distro-basic-smithi/5440500/